testnet/stacks-node/src/neon_node.rs

```diff
@@ -1406,27 +1387,34 @@ impl BlockMinerThread {
     /// Get the microblock private key we'll be using for this tenure, should we win.
     /// Return the private key on success
     /// return None if we were unable to generate the key.
-    fn make_microblock_private_key(&mut self, attempt: u64) -> Option<Secp256k1PrivateKey> {
+    ///
+    /// In testing, we ignore the parent stacks block hash because we don't have an easy way to
+    /// reproduce it in integration tests.
+    #[cfg(test)]
+    fn make_microblock_private_key(
+        &mut self,
+        _parent_stacks_hash: &StacksBlockId,
+    ) -> Secp256k1PrivateKey {
         // Generates a new secret key for signing the trail of microblocks
         // of the upcoming tenure.
-        let microblock_secret_key = if attempt > 1 {
-            match self.keychain.get_microblock_key() {
-                Some(k) => k,
-                None => {
-                    error!(
-                        "Relayer: Failed to obtain microblock key for mining attempt";
-                        "attempt" => %attempt
-                    );
-                    return None;
-                }
-            }
-        } else {
-            // NOTE: this is a no-op if run in a separate thread with a moved copy of the keychain
-            self.keychain
-                .rotate_microblock_keypair(self.burn_block.block_height)
-        };
+        self.keychain.make_microblock_secret_key(
+            self.burn_block.block_height,
+            &self.burn_block.block_height.to_be_bytes(),
+        )
+    }
 
-        Some(microblock_secret_key)
+    /// Get the microblock private key we'll be using for this tenure, should we win.
+    /// Return the private key on success
+    /// return None if we were unable to generate the key.
+    #[cfg(not(test))]
+    fn make_microblock_private_key(
+        &mut self,
+        _parent_stacks_hash: &StacksBlockId,
+    ) -> Secp256k1PrivateKey {
+        // Generates a new secret key for signing the trail of microblocks
+        // of the upcoming tenure.
+        self.keychain
+            .make_microblock_secret_key(self.burn_block.block_height, &[])
```

Comment:
Should the `parent_stacks_hash` be used for the salt here?

Reply:
In the test code, we don't do this because it's very hard to get this information from the integration test framework.

Reply:
Isn't this the non-test version? `#[cfg(not(test))]`